src/core/ddsc/src/dds_reader.c

```diff
@@ -658,14 +658,11 @@ static dds_entity_t dds_create_reader_int (dds_entity_t participant_or_subscribe
     iox_sub_options_t opts;
     iox_sub_options_init(&opts);
     
-    iox_sub_storage_extension_init(&rd->m_iox_sub_stor);
+    iox_sub_context_init(&rd->m_iox_sub_context);
 
     assert (rqos->durability.kind == DDS_DURABILITY_VOLATILE);
 
-    // TODO: need the max from iceoryx here,
-    // available in the API of master but not in release_1.0, will be available
-    // in the upcoming release 2.0 (planned March 2022)
-    const int32_t max_sub_queue_capacity = 256; // value from iceoryx 1.0
+    const uint32_t max_sub_queue_capacity = iox_cfg_max_subscriber_queue_capacity();
```

Comment:
It is unfortunate that we do not have a compile time constant here.This is by design, we have internal C++ constexpr but those cannot be propagated to C.

We would need `#defines`, but we want to avoid have two separate constants. This would require to derive the constexpr from a `#define`, which is not the C++ way ...